virtcontainers/mount.go

```diff
@@ -89,7 +89,9 @@ var errMountPointNotFound = errors.New("Mount point not found")
 //		manor : minor(/dev/sda1)
 //		mountPoint: /a/b/c
 //	}
-func getDeviceForPath(path string) (device, error) {
+var getDeviceForPath = findDeviceForPath
```

Comment:
@ganeshmaharaj cute but ouch.

@jodh-intel I see the value in adding this unit test. But instead of doing it this way, it may be better to just enhance our system level tests to include this path. Using devicemapper vs snapshotter.

Reply:
agree with @mcastelino ..we are going down the hole of adding too many indirections.
@ganeshmaharaj Is is possible to test this by actually creating an img and mounting it on a loop device?
Something like https://github.com/kata-containers/runtime/blob/6431f1f28883f7f60548e2651d68a6dd6aa7dc34/virtcontainers/container_test.go#L139 ?

Reply:
@amshinde I did think about it and I thought it was an overkill (creating a drive to just test this). But if you think it's worth it, I can hack this function to allow a mount point be passed and then handle the rootfs scenario.

Reply:
@mcastelino @amshinde fixed the patch to not hack any more files. Let me know how this one looks.

Reply:
lgtm @ganeshmaharaj